nav2_collision_monitor/src/detector_node.cpp

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) 2022 Samsung R&D Institute Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/detector_node.hpp"
+
+#include <exception>
+#include <utility>
+#include <functional>
+
+#include "tf2_ros/create_timer_ros.h"
+
+#include "nav2_util/node_utils.hpp"
+
+#include "nav2_collision_monitor/kinematics.hpp"
+
+using namespace std::chrono_literals;
+
+namespace nav2_collision_monitor
+{
+
+Detector::Detector(const rclcpp::NodeOptions & options)
+: nav2_util::LifecycleNode("detector", "", options),
+  process_active_(false)
+{
+}
+
+Detector::~Detector()
+{
+  polygons_.clear();
+  sources_.clear();
+}
+
+nav2_util::CallbackReturn
+Detector::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Configuring");
+
+  // Transform buffer and listener initialization
+  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(this->get_clock());
+  auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+    this->get_node_base_interface(),
+    this->get_node_timers_interface());
+  tf_buffer_->setCreateTimerInterface(timer_interface);
+  tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+
+  // Creating timer
+  timer_ = this->create_wall_timer(
```

Comment:
Move this to on activate + `timer_.reset()` on deactivate. Then you can remove the `process_active_` variable since `process()` could literally not be called if not active.